### PR TITLE
For #7718 - Unify BrowserToolbar and EngineView behaviors

### PR DESCRIPTION
**Pull request by Mugurell**
Allow clients to offer the same UX in regards to the dynamic toolbar feature
offered by our `BrowserToolbar` and `EngineView` when used together.
Going further this will ease development and ensure the best UX for users.

With this changes the bottom/top toolbars should function the same. Just mirrored.

image

[video showing the result](https://drive.google.com/file/d/1xYZZDxepoZ09xCFJScFTVa3Ijfp_rzsn/view?usp=sharing)

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBehavior.kt [hunk 1]
```diff
@@ -37,27 +39,20 @@ private const val SMALL_ELEVATION_CHANGE = 0.01f
  * - Snap the [BrowserToolbar] to be hidden or visible when the user stops scrolling.
  */
 @Suppress("TooManyFunctions")
-class BrowserToolbarBottomBehavior(
+class BrowserToolbarBehavior(
     val context: Context?,
-    attrs: AttributeSet?
+    attrs: AttributeSet?,
+    private val toolbarPosition: ToolbarPosition
```
Comment (pocmo) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBehavior.kt:45]: With that I assume you can no longer use this behavior from XML? But it's also not important to keep. :)

**Review by pocmo — APPROVED** [review]

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBehavior.kt (hunk 1 above)
Comment (Mugurell) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBehavior.kt:45]: Yes, currently we don't use it in xml so I skipped this part.
I could add an attribute for it later if needed?
Having in one place the code for both toolbars was the most important thing for me.

Comment (pocmo) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/behavior/BrowserToolbarBehavior.kt:45]: 👍

**Comment by Mugurell** [comment]
> fatal: unable to access 'https://github.com/Mugurell/android-components/': Failed to connect to github.com port 443: Connection refused

Let's try again

**Pull request merged**